src/domain-redirect.ts

```diff
@@ -0,0 +1,81 @@
+import * as acm from '@aws-cdk/aws-certificatemanager';
+import * as cf from '@aws-cdk/aws-cloudfront';
+import * as origins from '@aws-cdk/aws-cloudfront-origins';
+import * as route53 from '@aws-cdk/aws-route53';
+import * as route53_targets from '@aws-cdk/aws-route53-targets';
+import * as s3 from '@aws-cdk/aws-s3';
+import { Construct as CoreConstruct } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+
+/**
+ * Props for `DomainRedirect`.
+ */
+export interface DomainRedirectProps {
+  /**
+   * Source hosted zone.
+   */
+  readonly source: DomainRedirectSource;
+
+  /**
+   * The domain name to redirect to (e.g. `foo.com`).
+   */
+  readonly targetDomainName: string;
+}
+
+/**
+ * Redirects one domain to another domain using S3 and CloudFront.
+ */
+export class DomainRedirect extends CoreConstruct {
+  constructor(scope: Construct, id: string, props: DomainRedirectProps) {
+    super(scope, id);
+
+    const bucket = new s3.Bucket(this, 'Bucket', {
+      websiteRedirect: {
+        hostName: props.targetDomainName,
+      },
+    });
+
+    const sourceDomainName = props.source.hostedZone.zoneName;
+
+    const cert = props.source.certificate ?? new acm.DnsValidatedCertificate(this, 'Certificate', {
+      domainName: sourceDomainName,
+      hostedZone: props.source.hostedZone,
+    });
+
+    const dist = new cf.Distribution(this, 'Distribution', {
+      domainNames: [sourceDomainName],
+      defaultBehavior: { origin: new origins.S3Origin(bucket) },
+      certificate: cert,
+    });
```

Comment:
In order to comply with ORR requirements around CloudFront Distributions:

```suggestion
    const dist = new cf.Distribution(this, 'Distribution', {
      domainNames: [sourceDomainName],
      defaultBehavior: { origin: new origins.S3Origin(bucket) },
      certificate: cert,
      minimumProtocolVersion: cf.SecurityPolicyProtocol.TLS_V1_2_2018,
    });
```

Reply:
Additionally - I think this will only redirect the domain's root URL (`https://alternate.domain.name/`), but I feel this could be extended to also support redirecting arbitrary pages, too (`https://alternate.domain.name/packages/@aws-cdk/core/v/1.100.0`).

Reply:
Seems like it's redirecting all URLs. Check this out:

> https://benisrae.people.aws.dev/packages/@cdktf/provider-google/v/0.3.180?lang=typescript

Reply:
Oh neat!